### PR TITLE
Always emit seconds when formatting DATETIME and TIME values

### LINKED ISSUE #350: AvroToProtoSerializer produces invalid BigQuery DATETIME strings for values on exact minute boundaries (LocalDateTime.toString() omits the seconds)
`convertDateTime()` in `AvroToProtoSerializer` converts avro `local-timestamp-micros` / `local-timestamp-millis` values via `LocalDateTime.toString()` (both flink-1.17 and flink-2.1 modules on main; identical in releases 1.1.0 and 1.2.0). Java's `LocalDateTime.toString()` emits the shortest ISO-8601 form and omits the seconds when they are zero, so a value that lands exactly on a minute boundary is serialized as `"2026-08-31T12:30"`. BigQuery's DATETIME parser requires seconds and rejects the row:

```
INVALID_ARGUMENT: Invalid date time value '2026-08-31T12:30' on field '...'
{ error_code: CONVERTER_FIELD_VALUE_ERROR component: CONVERTER }
```

With the exactly-once (buffered stream) sink this is a deterministic poison row: the append is rejected on every retry, no checkpoint can ever complete, and the job enters a restart loop. Minute-boundary timestamps are legitimate data, so this cannot be avoided upstream of the sink. Whether it triggers depends only on the precision distribution of the source timestamps (any value with `micros % 60_000_000 == 0`).

The string-input branch of the same method has the same defect: `LocalDateTime.parse(value, DATETIME_FORMATTER).toString()` strips the seconds from an otherwise valid input string.

The TIME conversion has the same defect for string input: `LocalTime.parse(value, TIME_FORMATTER).toString()` turns `"12:30:00"` into `"12:30"`. The TIME conversion from a long already uses an explicit ISO formatter and is fine.

The existing `AvroToProtoSerializerTest` cases for DATETIME all use values with non-zero seconds, which is why the tests do not catch it.

**Self-contained reproduction** (no BigQuery access needed; depends only on the connector artifact and avro):

```java
import com.google.cloud.flink.bigquery.sink.serializer.AvroToProtoSerializer;
import com.google.cloud.flink.bigquery.sink.serializer.BigQuerySchemaProvider;
import com.google.cloud.flink.bigquery.sink.serializer.BigQuerySchemaProviderImpl;
import com.google.protobuf.ByteString;
import com.google.protobuf.DynamicMessage;
import org.apache.avro.Schema;
import org.apache.avro.generic.GenericData;
import org.apache.avro.generic.GenericRecord;

public class DatetimeMinuteBoundaryRepro {
    static final String SCHEMA_JSON =
            "{\"type\":\"record\",\"name\":\"Row\",\"fields\":["
                    + "{\"name\":\"dt\",\"type\":{\"type\":\"long\",\"logicalType\":\"local-timestamp-micros\"}}]}";

    public static void main(String[] args) throws Exception {
        Schema avro = new Schema.Parser().parse(SCHEMA_JSON);
        BigQuerySchemaProvider provider = new BigQuerySchemaProviderImpl(avro);
        AvroToProtoSerializer serializer = new AvroToProtoSerializer();
        serializer.init(provider);
        long minuteBoundary = 1_788_179_400_000_000L; // 2026-08-31T12:30:00.000000
        long ordinary = 1_788_179_407_123_456L; // 2026-08-31T12:30:07.123456
        for (long micros : new long[] {minuteBoundary, ordinary}) {
            GenericRecord r = new GenericData.Record(avro);
            r.put("dt", micros);
            ByteString bytes = serializer.serialize(r);
            DynamicMessage msg = DynamicMessage.parseFrom(provider.getDescriptor(), bytes);
            System.out.println("micros=" + micros + "  ->  proto " + msg.toString().trim());
        }
    }
}
```

Output:

```
micros=1788179400000000  ->  proto dt: "2026-08-31T12:30"
micros=1788179407123456  ->  proto dt: "2026-08-31T12:30:07.123456"
```

The first string is rejected by BigQuery (`SELECT DATETIME '2026-08-31T12:30'` fails with `Invalid DATETIME literal` as well).

**Suggested fix:** format with explicit patterns that always emit seconds (`uuuu-MM-dd'T'HH:mm:ss.SSSSSS` for DATETIME, `HH:mm:ss.SSSSSS` for TIME) in both the epoch and the string branches. This is the same class of bug that #258 fixed for the filter pushdown literals in `BigQueryRestriction`, which already uses exactly these two patterns.

**Workaround until a fix is released:** keep the BigQuery column as DATETIME but declare the avro field as a plain `string` (the proto field is a string either way) and format the value in the pipeline; the serializer then passes the string through without the logical-type conversion.

A PR with the formatter change and minute-boundary tests for both the flink-1.17 and flink-2.1 modules follows.

### PR DESCRIPTION
**Pull request by seokjin0414**
Fixes #350

`AvroToProtoSerializer` converted avro `local-timestamp-micros` / `local-timestamp-millis` values to BigQuery DATETIME strings with `LocalDateTime.toString()`, and TIME strings from string input with `LocalTime.toString()`. Both print the shortest ISO-8601 form and omit the seconds when they are zero, so a value on an exact minute boundary was serialized as `"2026-08-31T12:30"`, which BigQuery rejects (`INVALID_ARGUMENT: Invalid date time value '2026-08-31T12:30'`, `CONVERTER_FIELD_VALUE_ERROR`). With the exactly-once sink this is a deterministic poison row: the append is rejected on every retry, no checkpoint completes, and the job restarts forever. The string input branch had the same defect and stripped the seconds from an otherwise valid input.

This is the same class of bug that #258 fixed for the filter pushdown literals in `BigQueryRestriction`, and the fix uses the same two patterns. They now live in `DateTimeFormatterPatterns` so that both call sites share them.

## Summary

- `DateTimeFormatterPatterns` gains `DATETIME_OUTPUT_FORMATTER` (`uuuu-MM-dd'T'HH:mm:ss.SSSSSS`) and `TIME_OUTPUT_FORMATTER` (`HH:mm:ss.SSSSSS`).
- `convertDateTime` uses it in both the epoch and the string branch; `convertTime` uses it in the string branch. The TIME conversion from a long already used `DateTimeFormatter.ISO_TIME`, which always emits the seconds; it is unchanged and gets a regression test.
- `BigQueryRestriction` points its private formatter constants at the shared ones instead of its own copies of the same patterns, and its `INSTANT_FORMATTER` uses `uuuu` as well; `BigQueryRestrictionTest` is unchanged.
- Fractional seconds are now always six digits, the precision of DATETIME and TIME and the width #258 already settled on, so the existing expectations with three digits (`.347`, `.462`) are updated to `.347000` / `.462000`. Values with millisecond or whole-second precision serialize 3 to 7 bytes longer than before; microsecond values are unchanged.
- New tests cover minute-boundary values, with and without a fraction, for the long, millisecond and string inputs (DATETIME) and for the string and long inputs (TIME). All DATETIME cases and the TIME string case fail on the current code (`2026-08-31T12:30` and `12:30`); the TIME long case passes on the current code and guards the existing `ISO_TIME` behavior.

Applied to both `flink-1.17-connector-bigquery` and `flink-2.1-connector-bigquery`; the Table API path (`RowDataToProtoSerializer`) delegates to the same conversion.

## Testing

- `./mvnw -Pflink_1.17 -pl flink-1.17-connector-bigquery/flink-connector-bigquery -am -Dtest=AvroToProtoSerializerTest,RowDataToProtoSerializerTest,BigQueryRestrictionTest -Dsurefire.failIfNoSpecifiedTests=false -DfailIfNoTests=false test`
- `./mvnw -Pflink_2.1 -pl flink-2.1-connector-bigquery/flink-connector-bigquery -am -Dtest=AvroToProtoSerializerTest,RowDataToProtoSerializerTest,BigQueryRestrictionTest -Dsurefire.failIfNoSpecifiedTests=false -DfailIfNoTests=false test`
- `./mvnw -Pflink_1.17 spotless:check checkstyle:check` and the same with `-Pflink_2.1`
- The new DATETIME cases and the TIME string case fail on the current code (`2026-08-31T12:30` and `12:30`).
- Full unit test suites of both connector modules pass.

### CODE REVIEW

File: flink-connector-bigquery-common/src/main/java/com/google/cloud/flink/bigquery/common/utils/DateTimeFormatterPatterns.java [hunk 1]
```diff
@@ -59,4 +59,20 @@ private DateTimeFormatterPatterns() {}
...
+     * Formatter for DATETIME values sent to BigQuery, both as sink payload strings and as literals
+     * in pushed-down filters.
+     *
+     * <p>BigQuery requires the seconds to be present, but {@link
+     * java.time.LocalDateTime#toString()} omits them when they are zero (for example
+     * "2024-03-13T00:42" for a value on a minute boundary), so values are always written as
+     * "uuuu-MM-dd'T'HH:mm:ss.SSSSSS" with the six fractional digits BigQuery stores.
+     */
+    public static final DateTimeFormatter DATETIME_OUTPUT_FORMATTER =
+            DateTimeFormatter.ofPattern("uuuu-MM-dd'T'HH:mm:ss.SSSSSS");
```
[the remark this answers is not in the document]
Comment (seokjin0414) [reply flink-connector-bigquery-common/src/main/java/com/google/cloud/flink/bigquery/common/utils/DateTimeFormatterPatterns.java]: Done in dc924ac: the three formatters are now built with Locale.ROOT. These patterns contain only numeric fields and literals, and DateTimeFormatter.ofPattern uses the standard decimal style regardless of the default locale, so the output is unchanged; the argument makes that explicit.
